cmd/ris-mirror/rismirror/router.go

```diff
@@ -54,15 +56,38 @@ func (r *Router) GetVRFs() []*vrf.VRF {
 func (r *Router) addVRF(rd uint64, sources []*grpc.ClientConn) {
 	v := r.vrfRegistry.CreateVRFIfNotExists(vrf.RouteDistinguisherHumanReadable(rd), rd)
 
-	r.vrfs[rd] = newVRF(v.IPv4UnicastRIB(), v.IPv6UnicastRIB())
+	r.vrfs[rd] = newVRF(v)
 
+	// cleanup old ris clients of the vrf
+	r.stopVRF(rd)
```

Comment:
Why do we need to stop a freshly created VRF?

Reply:
Now we have a deadlock here. `vrfClientsMu` gets locked here and also in `stopVRF()`.

Reply:
Acquiring the lock once and then doing all the work is the right thing here. We need a `stopVRF()` function that doesn't lock on its own. Create `_stopVRF()` that does the same without the lock. Change `stopVRF()` to only do locking and call `_stopVRF()`.